lib/omniauth/strategies/openid_connect.rb

```diff
@@ -132,6 +137,13 @@ def other_phase
           return redirect(end_session_uri) if end_session_uri
         end
 
+        if options.backchannel_logout_callback && backchannel_logout_path_pattern.match?(current_path)
+          discover!
+          perform_backchannel_logout!(params['logout_token'])
+          return Rack::Response.new([''], 204).finish
```

Comment:
As per the specification in section 2.8 (https://openid.net/specs/openid-connect-backchannel-1_0.html) this MUST return 200 rather than 204. It also says that an OP needs to be prepared for a 204 but we might as well do it right. Or is there a specific reason we use 204 here (other than the usual convention of 204 being returned for an empty body)?

> If the logout succeeded, the RP MUST respond with HTTP 200 OK

In the same vein, the specification says 400 must be returned if the logout failed. But right now we do not actually check if it did. Should we add this? This would mean evaluating a return value for the logout callback.

> If the logout request was invalid or the logout failed, the RP MUST respond with HTTP 400 Bad Request.

Reply:
Also it is suggested to include `Cache-Control: no-store` in the response. We might as well add that too.

> The RP's response SHOULD include the Cache-Control HTTP response header field with a no-store value, keeping the response from being cached to prevent cached responses from interfering with future logout requests. An example of this is:
> 
>  Cache-Control: no-store